doc/md/examples/grammar.txt

```diff
@@ -287,15 +284,15 @@
 
 <pat_field> ::= 
     <id> (':' <typ>)?
-    <id> <annot_opt_eq> <pat>
+    <id> (':' <typ>)? '=' <pat>
 
 <dec_var> ::= 
-    'var' <id> <annot_opt_eq> <exp>
+    'var' <id> (':' <typ>)? '=' <exp>
 
 <dec_nonvar> ::= 
     'let' <pat> '=' <exp>
     'type' <id> ('<' <list(<typ_bind>, ',')> '>')? '=' <typ>
-    <obj_sort> <id>? '='? <obj_body>
+    <obj_sort> <id>? (':' <typ>)? '='? <obj_body>
```

Comment:
this admits `: module {} {}` which I distaste :-(

Reply:
Oh, I see. Yeah, I guess that is consfusing.

Reply:
This form is valid too, so we can choose:
``` Motoko
> module class M() : module { a : Int } = { public let a = 42 };
type M__6 = module {a : Nat}
```